src/io/flutter/perf/VmServiceWidgetPerfProvider.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.perf;
+
+import com.google.gson.JsonObject;
+import com.intellij.openapi.fileEditor.FileEditor;
+import com.intellij.openapi.util.text.StringUtil;
+import io.flutter.run.daemon.FlutterApp;
+import io.flutter.utils.StreamSubscription;
+import io.flutter.utils.VmServiceListenerAdapter;
+import org.dartlang.vm.service.VmService;
+import org.dartlang.vm.service.VmServiceListener;
+import org.dartlang.vm.service.consumer.ServiceExtensionConsumer;
+import org.dartlang.vm.service.element.Event;
+import org.dartlang.vm.service.element.IsolateRef;
+import org.dartlang.vm.service.element.RPCError;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+
+public class VmServiceWidgetPerfProvider implements WidgetPerfProvider {
```

Comment:
@devoncarew  this code should generally look familiar although I did a bit of a quick and dirty job to refactor it out for testability.